workflows/preprocessing.nf

```diff
@@ -134,6 +135,34 @@ workflow PREPROCESSING {
     }
     .set {ch_input_fastq}
 
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// STEP: UMI CONSENSUS (optional, before alignment)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+UMI_FLAG = Channel.value(params.umi_in_readname as boolean)
+if (params.enable_umi) {
+
+    // Convert to [meta, r1, r2] for the UMI subworkflow
+    CH_UMI_FASTQ = ch_input_fastq.map { meta, reads ->
+        def r1 = (reads instanceof List) ? reads[0] : reads
+        def r2 = (reads instanceof List && reads.size() > 1) ? reads[1] : null
```

Comment:
```suggestion
        def r2 = (reads instanceof List && reads.size() > 1) ? reads[1] : []
```
Try to always set `[]` as the `null` value for files, otherwise these will not be considered a valid input option later